I do like the new test, now there's just the little detail that it's not quite working yet

```
[INFO] Running org.apache.hadoop.io.TestSequenceFile
[ERROR] Tests run: 10, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 9.818 s <<< FAILURE! - in org.apache.hadoop.io.TestSequenceFile
[ERROR] testSequenceFileWriter(org.apache.hadoop.io.TestSequenceFile)  Time elapsed: 0.614 s  <<< ERROR!
java.io.IOException: wrong key class: org.apache.hadoop.io.LongWritable is not class org.apache.hadoop.io.NullWritable
	at org.apache.hadoop.io.SequenceFile$RecordCompressWriter.append(SequenceFile.java:1508)
	at org.apache.hadoop.io.SequenceFile$Writer.append(SequenceFile.java:1425)
	at org.apache.hadoop.io.TestSequenceFile.testSequenceFileWriter(TestSequenceFile.java:745)
```

+ minor checkstyles

```
./hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java:737:    Path p = new Path(GenericTestUtils.getTempPath("testSequenceFileWriter.seq"));: Line is longer than 80 characters (found 82). [LineLength]
./hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java:745:      writer.append(key,value);:24: ',' is not followed by whitespace. [WhitespaceAfter]
```

--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java
@@ -747,7 +747,7 @@ public class TestSequenceFile {
       LongWritable key = new LongWritable();
       key.set(1);
       Text value = new Text();
-      value.set("value");
+      value.set("somevalue");
       writer.append(key, value);
       writer.flush();
       writer.hflush();

--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java
@@ -734,15 +734,16 @@ public class TestSequenceFile {
   @Test
   public void testSequenceFileWriter() throws Exception {
     FileSystem fs = FileSystem.getLocal(conf);
-    Path p = new Path(GenericTestUtils.getTempPath("testSequenceFileWriter.seq"));
+    Path p = new Path(GenericTestUtils
+      .getTempPath("testSequenceFileWriter.seq"));
     try(SequenceFile.Writer writer = SequenceFile.createWriter(
-            fs, conf, p, NullWritable.class, NullWritable.class)) {
+            fs, conf, p, LongWritable.class, Text.class)) {
 
       LongWritable key = new LongWritable();
       key.set(1);
       Text value = new Text();
       value.set("value");
-      writer.append(key,value);
+      writer.append(key, value);
       writer.hflush();
       writer.hsync();
       Assertions.assertThat(fs.getFileStatus(p).getLen()).isGreaterThan(0);

--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/TestSequenceFile.java
@@ -733,17 +733,23 @@ public class TestSequenceFile {
 
   @Test
   public void testSequenceFileWriter() throws Exception {
-    FileSystem fs = FileSystem.getLocal(conf);
+    Configuration conf = new Configuration();
+    // This test only works with Raw File System and not Local File System
+    FileSystem fs = FileSystem.getLocal(conf).getRaw();
     Path p = new Path(GenericTestUtils
       .getTempPath("testSequenceFileWriter.seq"));
     try(SequenceFile.Writer writer = SequenceFile.createWriter(
             fs, conf, p, LongWritable.class, Text.class)) {
-
+      Assertions.assertThat(writer.hasCapability
+        (StreamCapabilities.HSYNC)).isEqualTo(true);
+      Assertions.assertThat(writer.hasCapability(
+        StreamCapabilities.HFLUSH)).isEqualTo(true);
       LongWritable key = new LongWritable();
       key.set(1);
       Text value = new Text();
       value.set("value");
       writer.append(key, value);
+      writer.flush();
       writer.hflush();
       writer.hsync();
       Assertions.assertThat(fs.getFileStatus(p).getLen()).isGreaterThan(0);

